frontends/p4/tableKeyNames.cpp

```diff
@@ -32,10 +38,16 @@ class KeyNameGenerator : public Inspector {
 
     void postorder(const IR::Expression* expression) override { error(expression); }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("foo.bar")` for
```

Comment:
I don't imply that the work is useless, that particular comment is useless. That is different.

Also, there is such a thing as over-commenting; we have no tools to keep comments in sync with code, and a wrong comment is worse than no comment. So I think we should really write comments where they really help our understanding of the code/algorithms. Sometimes code is clearer than comments.

I don't think that uniform documentation for Doxygen is a worthwhile goal.